### ISSUE TITLE
Sep Pay Per Use – Usage Tracking and Charging Infrastructure

### ISSUE THREAD

**Issue by elisabettai** [ITISFoundation/osparc-issues#1103]
Description:
In the future, it will be necessary for sustainability purposes to attribute (and charge) resource usage costs to commercial users, users exceeding the limits of the free usage tier, or whatever the adopted sustainability model foresees. This potentially includes compute time, simulation hours and S3 storage per user, as well as egress costs for download of files through the API. A cost explorer with detailed information about spent simulation hours (see DevOps infrastructure & automation regarding the collection of such information) and remaining credits will be established. This functionality can in the future be linked to a website to purchase additional simulation hours.

Deliverable: 
Usage tracking and charging Infrastructure

Acceptance criteria: 
User resource usage can be tracked and presented to the user, and infrastructure is in place to permit charging users for their resource costs (likely in the form of prepayment, i.e., buying credit) 

Out of scope: 
Webpage for purchasing resources

Deadline:
Q1

[wrike Y7-MS 7.2: Pay Per Use – Usage Tracking and Charging Infrastructure](https://www.wrike.com/open.htm?id=1155714992)

**Comment by elisabettai** [comment]
@pcrespov, this is related to the milestone for NIH.
Reading the "Acceptance criteria", I'd say that we have resource usage well tracked and presented to the user, and we're able to attribute/charge users (as we're doing since the internal launch).

Do you still see any important detail missing to consider this completed?
I checked the WP7, and this seems the only detail missing: https://github.com/ITISFoundation/osparc-simcore/pull/5057. Is this very critical for the overall functionality?

**Comment by pcrespov** [comment]
> Do you still see any important detail missing to consider this completed? I checked the related [WP10](https://github.com/ITISFoundation/osparc-issues/issues/1027), and this seems the only detail missing: [ITISFoundation/osparc-simcore#5057](https://github.com/ITISFoundation/osparc-simcore/pull/5057). Is this very critical for the overall functionality?

@elisabettai Yes, I agree. This is completed. The issue you point is a last enhancement and will in addition be closed in the coming days.
